StitchCore-iOS/StitchCore-iOS/Core/Auth/Internal/StitchAuthImpl.swift

```diff
@@ -26,78 +52,172 @@ internal final class StitchAuthImpl: CoreStitchAuth<StitchUserImpl>, StitchAuth
         self.appInfo = appInfo
         try super.init(requestClient: requestClient, authRoutes: authRoutes, storage: storage)
     }
-    
-    public func providerClient<Provider>(forProvider provider: Provider) -> Provider.Client where Provider : AuthProviderClientSupplier {
-        return provider.client(withRequestClient: self.requestClient, withRoutes: self.authRoutes, withDispatcher: self.dispatcher)
+
+    // MARK: Authentication Provider Clients
+
+    /**
+     * Retrieves the authentication provider client associated with the authentication provider type specified in the
+     * argument.
+     *
+     * - parameters:
+     *     - forProvider: The authentication provider conforming to `AuthProviderClientSupplier` which will provide the
+     *                    client for this authentication provider. Use the `clientProvider` field of the desired
+     *                    authentication provider class.
+     * - returns: an authentication provider client whose type is determined by the `Client` typealias in the type
+     *            specified in the `forProvider` parameter.
+     */
+    public func providerClient<Provider>(forProvider provider: Provider)
+        -> Provider.Client where Provider: AuthProviderClientSupplier {
```

Comment:
[q] It's called a `AuthProviderClientSupplier` but the instances are a `clientProvider`. Should we change one or the other?

Reply:
I'll rename to `clientSupplier`. I think `Provider` is confusing because of the fact that it's for an authentication provider.